src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -13,6 +13,40 @@ const MISSING_BLOCK_ERROR_MESSAGE = 'The "block" field type needs at least one t
 const BLOCK_PREVIEW_TAG = 'sulu.block_preview';
 
 export default class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
+    componentDidUpdate(prevProps: FieldTypeProps<Array<BlockEntry>>) {
+        const {defaultType, types, value: value, onChange} = this.props;
+        const {types: oldTypes} = prevProps;
+
+        if (!types || !oldTypes) {
+            throw new Error(MISSING_BLOCK_ERROR_MESSAGE);
+        }
+
+        const newValue = toJS(value);
+        let hasValueChanged = false;
```

Comment:
Is there a reason you have done it now in a different way than I suggested in [my comment](https://github.com/sulu/sulu/pull/5099#discussion_r394349791)? I.e. did you have to use the `let` here now?

Reply:
First to make it easier extend so if somebody change in componentDidUpdate the value it just need to set that boolean flag to true to call onChange and to avoid a second loop over all blocks.

Reply:
Ok, that makes sense, especially since if we end up with two `onChange` calls in here one would override the other changes.

But couldn't we do this without this variable? I would just check if the old value is the same as the new value using `equals` as we do in other places. With this implementation the `hasValueChanged` flag might have been set although the `value` has not changed. If we keep all of that in a single variable, it is guaranteed to be correct.